### PR TITLE
[SPARK-25167][SPARKR][TEST][MINOR] Minor fixes for R sql tests

### PR DESCRIPTION
**Pull request by dilipbiswal**
## What changes were proposed in this pull request?
A few SQL tests for R were failing in my development environment. In this PR, i am attempting to
address some of them.  Below are the reasons for the failure.

- The catalog api tests assumes catalog artifacts named "foo" to be non existent. I think name such as foo and bar are common and i use it frequently. I have changed it to a string that i hope is less likely to collide.
- One test assumes that we only have one database in the system. I had more than one and it caused the test to fail. I have changed that check.
- One more test which compares two timestamp values fail - i am debugging this now. I will send it as a followup - may be.
   
## How was this patch tested?
Its a test fix.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Eh, @dilipbiswal, actually can we file a JIRA?

**Comment by dilipbiswal** [comment]
@HyukjinKwon Done. [SPARK-25167](https://issues.apache.org/jira/browse/SPARK-25167)

File: R/pkg/tests/fulltests/test_sparkSQL.R [hunk 1]
```diff
@@ -3613,11 +3613,11 @@ test_that("Collect on DataFrame when NAs exists at the top of a timestamp column
...
   expect_equal(currentDatabase(), "default")
   expect_error(setCurrentDatabase("default"), NA)
-  expect_error(setCurrentDatabase("foo"),
-               "Error in setCurrentDatabase : analysis error - Database 'foo' does not exist")
+  expect_error(setCurrentDatabase("zxwtyswklpf"),
+        "Error in setCurrentDatabase : analysis error - Database 'zxwtyswklpf' does not exist")
   dbs <- collect(listDatabases())
   expect_equal(names(dbs), c("name", "description", "locationUri"))
-  expect_equal(dbs[[1]], "default")
+  expect_equal(which(dbs[, 1] == "default"), 1)
```
Comment (felixcheung) [inline R/pkg/tests/fulltests/test_sparkSQL.R]: I wonder if there is a better way to ensure the default database is named "default", perhaps? this checks for "exactly one database is named "default"" - I guess that's ok...

**Review by felixcheung — APPROVED** [review]

File: R/pkg/tests/fulltests/test_sparkSQL.R (hunk 1 above)
Comment (dilipbiswal) [reply R/pkg/tests/fulltests/test_sparkSQL.R]: @felixcheung Thanks for reviewing. Actually the name "default" for default database is fixed, to the best of my knowledge. Its even hardcoded [here](https://github.com/apache/spark/blob/c3be2cd347c42972d9c499b6fd9a6f988f80af12/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L45-47). So we should be okay with this check. Please let me know what you think.

File: R/pkg/tests/fulltests/test_sparkSQL.R [hunk 2]
```diff
@@ -3640,17 +3640,18 @@ test_that("catalog APIs, listTables, listColumns, listFunctions", {
...
   f <- listFunctions()
   expect_true(nrow(f) >= 200) # 250
   expect_equal(colnames(f),
                c("name", "database", "description", "className", "isTemporary"))
   expect_equal(take(orderBy(f, "className"), 1)$className,
                "org.apache.spark.sql.catalyst.expressions.Abs")
-  expect_error(listFunctions("foo_db"),
-               "Error in listFunctions : analysis error - Database 'foo_db' does not exist")
+  expect_error(listFunctions("zxwtyswklpf_db"),
+        "Error in listFunctions : analysis error - Database 'zxwtyswklpf_db' does not exist")
```
Comment (HyukjinKwon) [inline R/pkg/tests/fulltests/test_sparkSQL.R]: I think the style below is more correct:

```R
expect_error(listFunctions("zxwtyswklpf_db"),
             "Error in ...
```

**Comment by HyukjinKwon** [comment]
Seems okay to me.

File: R/pkg/tests/fulltests/test_sparkSQL.R [hunk 3]
```diff
@@ -3613,11 +3613,11 @@ test_that("Collect on DataFrame when NAs exists at the top of a timestamp column
 test_that("catalog APIs, currentDatabase, setCurrentDatabase, listDatabases", {
   expect_equal(currentDatabase(), "default")
   expect_error(setCurrentDatabase("default"), NA)
-  expect_error(setCurrentDatabase("foo"),
-               "Error in setCurrentDatabase : analysis error - Database 'foo' does not exist")
+  expect_error(setCurrentDatabase("zxwtyswklpf"),
+        "Error in setCurrentDatabase : analysis error - Database 'zxwtyswklpf' does not exist")
```
Comment (HyukjinKwon) [inline R/pkg/tests/fulltests/test_sparkSQL.R]: Not a big deal but can we fix this style too?

**Comment by HyukjinKwon** [comment]
LGTM. 

Merged to master.

**Comment by dilipbiswal** [comment]
@HyukjinKwon Oh.. thank you. I was going to fix the style ? I will include it when i fix something next ?

**Comment by HyukjinKwon** [comment]
Ah, it's okie. Yes, please. Not a big deal.

**Pull request closed without merging**